CRM/Aoservicelisting/Form/ProviderApplication.php

```diff
@@ -89,6 +89,23 @@ public function preProcess() {
       $this->organizationId = $loggedInContactId;
       $this->set('organizationId', $loggedInContactId);
     }
+    if (!empty($this->organizationId) && !empty($this->_loggedInContactID)) {
+      // Check status to make sure we only allow edits to Approved service listings.
+      $allowEdit = FALSE;
+      $status = civicrm_api3('Contact', 'get', [
+        'id' => $this->organizationId,
+        'sequential' => 1,
+        'return' => [STATUS],
+      ]);
+      if (!empty($status['values'])) {
+        if ($status['values'][0][STATUS] == 'Approved') {
+          $allowEdit = TRUE;
+        }
+      }
+      if (empty($allowEdit)) {
+        CRM_Core_Error::statusBounce(ts('This application has already been submitted and is under progress.'), CRM_Utils_System::url('civicrm/service-listing-application', 'reset=1'));
```

Comment:
@Edzelopez can you please change the redirect url to something other than application page (may be homepage?) because the source and target url is same and thus it traps in a infinite redirecting loop. 

i.e. 
```
        CRM_Core_Error::statusBounce(ts('This application has already been submitted and is under progress.'), CRM_Utils_System::url('civicrm', 'reset=1'));
        ```